tests/testthat/test-detect.r

```diff
@@ -28,10 +28,18 @@ test_that("modifiers work", {
 test_that("str_starts works", {
   expect_true(str_starts("ab", "a"))
   expect_false(str_starts("ab", "b"))
+  expect_true(str_starts("ab", fixed("a")))
```

Comment:
I think you should only need 4 new tests — one each for `str_starts()` and `str_end()`, and `fixed` and `coll`